Sources/NIOHTTPTypesHTTP1/HTTPTypeConversion.swift

```diff
@@ -0,0 +1,203 @@
+        case .unlink: self = .UNLINK
+        case .connect: self = .CONNECT
+        case .msearch: self = .MSEARCH
+        case .options: self = .OPTIONS
+        case .propfind: self = .PROPFIND
+        case .checkout: self = .CHECKOUT
+        case .proppatch: self = .PROPPATCH
+        case .subscribe: self = .SUBSCRIBE
+        case .mkcalendar: self = .MKCALENDAR
+        case .mkactivity: self = .MKACTIVITY
+        case .unsubscribe: self = .UNSUBSCRIBE
+        case .source: self = .SOURCE
+        default: self = .RAW(value: newMethod.rawValue)
+        }
+    }
+
+    var newMethod: HTTPRequest.Method {
+        get throws {
+            switch self {
+            case .GET: return .get
+            case .PUT: return .put
+            case .ACL: return .acl
+            case .HEAD: return .head
+            case .POST: return .post
+            case .COPY: return .copy
+            case .LOCK: return .lock
+            case .MOVE: return .move
+            case .BIND: return .bind
+            case .LINK: return .link
+            case .PATCH: return .patch
+            case .TRACE: return .trace
+            case .MKCOL: return .mkcol
+            case .MERGE: return .merge
+            case .PURGE: return .purge
+            case .NOTIFY: return .notify
+            case .SEARCH: return .search
+            case .UNLOCK: return .unlock
+            case .REBIND: return .rebind
+            case .UNBIND: return .unbind
+            case .REPORT: return .report
+            case .DELETE: return .delete
+            case .UNLINK: return .unlink
+            case .CONNECT: return .connect
+            case .MSEARCH: return .msearch
+            case .OPTIONS: return .options
+            case .PROPFIND: return .propfind
+            case .CHECKOUT: return .checkout
+            case .PROPPATCH: return .proppatch
+            case .SUBSCRIBE: return .subscribe
+            case .MKCALENDAR: return .mkcalendar
+            case .MKACTIVITY: return .mkactivity
+            case .UNSUBSCRIBE: return .unsubscribe
+            case .SOURCE: return .source
+            case .RAW(value: let value):
+                guard let method = HTTPRequest.Method(value) else {
+                    throw HTTP1TypeConversionError.invalidMethod
+                }
+                return method
+            }
+        }
+    }
+}
+
+extension HTTPHeaders {
+    init(_ newFields: HTTPFields) {
+        let fields = newFields.map { ($0.name.rawName, $0.value) }
+        self.init(fields)
+    }
+
+    func newFields(splitCookie: Bool) -> HTTPFields {
+        var fields = HTTPFields()
+        fields.reserveCapacity(count)
+        for (index, field) in enumerated() {
+            if index == 0, field.name.lowercased() == "host" {
+                continue
+            }
+            if let name = HTTPField.Name(field.name) {
+                if splitCookie, name == .cookie, #available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *) {
```

Comment:
As a general style nit we prefer to use `&&` where it is possible to do so.

Reply:
This was actually changed to `,` by swiftformat. I changed it back.